tests/test_odp_manager.py

```diff
@@ -0,0 +1,299 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from unittest import mock
+
+from optimizely import exceptions as optimizely_exception
+from optimizely.helpers.enums import Errors
+from optimizely.odp.lru_cache import LRUCache
+from optimizely.odp.odp_config import OdpConfig
+from optimizely.odp.odp_event_manager import OdpEventManager
+from optimizely.odp.odp_manager import OdpManager
+from optimizely.odp.odp_segment_manager import OdpSegmentManager
+from optimizely.odp.zaius_graphql_api_manager import ZaiusGraphQLApiManager
+from optimizely.odp.zaius_rest_api_manager import ZaiusRestApiManager
+from tests import base
+
+
+class OdpManagerTest(base.BaseTest):
+    cache_size = 10
+    cache_timeout = 20
+
+    def test_configurations_disable_odp(self):
+        mock_logger = mock.MagicMock()
+
+        manager = OdpManager(True, 10, 20, None, None, mock_logger)
+        manager.fetch_qualified_segments('user1', [])
+        mock_logger.debug.assert_called_once_with(Errors.ODP_NOT_ENABLED)
+
+        manager.update_odp_config('valid', 'host', [])
+        self.assertIsNone(manager.odp_config.get_api_key())
+        self.assertIsNone(manager.odp_config.get_api_host())
+
+        # these call should be dropped gracefully with None
+        manager.identify_user('user1')
+
+        self.assertRaisesRegex(optimizely_exception.OdpNotEnabled, Errors.ODP_NOT_ENABLED,
+                               manager.send_event, 't1', 'a1', {}, {})
+
+        self.assertIsNone(manager.event_manager)
+        self.assertIsNone(manager.segment_manager)
+
+    def test_fetch_qualified_segments(self):
+        mock_logger = mock.MagicMock()
+        segment_manager = OdpSegmentManager(OdpConfig(), LRUCache(1000, 1000),
+                                            ZaiusGraphQLApiManager(), mock_logger)
+
+        manager = OdpManager(False, 10, 20, segment_manager, None, mock_logger)
+
+        with mock.patch.object(segment_manager, 'fetch_qualified_segments') as mock_fetch_qualif_segments:
+            manager.fetch_qualified_segments('user1', [])
+
+        mock_logger.debug.assert_not_called()
+        mock_fetch_qualif_segments.assert_called_once_with('fs_user_id', 'user1', [])
+
+    def test_identify_user_datafile_not_ready(self):
+        mock_logger = mock.MagicMock()
+        event_manager = OdpEventManager(OdpConfig(), mock_logger)
+        event_manager.start()
+
+        manager = OdpManager(False, 10, 20, None, event_manager, mock_logger)
+
+        with mock.patch.object(event_manager, 'identify_user') as mock_identify_user:
+            manager.identify_user('user1')
+
+        mock_identify_user.assert_called_once_with('user1')
+
+    def test_identify_user_odp_integrated(self):
+        mock_logger = mock.MagicMock()
+        event_manager = OdpEventManager(OdpConfig(), mock_logger, ZaiusRestApiManager())
+        event_manager.start()
```

Comment:
We actually want the OdpManager to start the event manager upon initialization.

Reply:
done.